File: src/main/java/ukalus/items/RandomWord.java
/*
 * Created on Feb 25, 2005
 *
 */
package ukalus.items;

import ukalus.util.Closure;

import java.util.Random;

class RandomWord implements Closure<Object, String> {

  private static final long serialVersionUID = 3258407339714425905L;
  private Random random;
  private String[] items;

  public RandomWord(Random random, String[] items) {
    this.random = random;
    this.items = items;
  }

  public String apply(Object argument) {
    return items[random.nextInt(items.length)];
  }
}

File: src/main/java/ukalus/items/RandomWord.kt
/*
 * Created on Feb 25, 2005
 *
 */
package ukalus.items

import ukalus.util.random
import java.util.Random
import java.util.function.Function

internal class RandomWord(private val random: Random, private val items: Array<String>) : Function<Any, String> {

    override fun apply(argument: Any) = items.random(random)!!
}